src/Core/WinSWCore/Configuration/YamlConfiguration.cs

```diff
@@ -0,0 +1,203 @@
+
+        [YamlMember(Alias = "serviceaccount")]
+        public ServiceAccount? ServiceAccount { get; set; }
+        string? IWinSWConfiguration.ServiceAccountPassword => ServiceAccount != null ? ServiceAccount.password : null;
+        string? IWinSWConfiguration.ServiceAccountUser => ServiceAccount != null ? ServiceAccount.user : null;
+
+        [YamlMember(Alias = "log")]
+        public YAMLLog? _YAMLLog { get; set; }
+        public Log? Log => _YAMLLog;
+
+        string IWinSWConfiguration.LogDirectory => Log != null ? Log.Directory : null;
+        string IWinSWConfiguration.LogMode => Log != null ? Log.Directory : null;
+
+        
+
+
+        [YamlMember(Alias = "download")]
+        public List<Download> Downloads { get; set; }
+
+        [YamlMember(Alias = "caption")]
+        public string? Caption { get; set; }
+
+        [YamlMember(Alias = "executablePath")]
+        public string? ExecutablePath { get; set; }
+
+        [YamlMember(Alias = "hideWindow")]
+        public bool HideWindow { get; set; }
+
+        [YamlMember(Alias = "allowServiceAcountLogonRight")]
+        public bool AllowServiceAcountLogonRight { get; set; }
+
+        [YamlMember(Alias = "arguments")]
+        public string? Arguments { get; set; }
+
+        [YamlMember(Alias = "startArguments")]
+        public string? StartArguments { get; set; }
+
+        [YamlMember(Alias = "stopExecutable")]
+        public string? StopExecutable { get; set; }
+
+        [YamlMember(Alias = "stopArguments")]
+        public string? StopArguments { get; set; }
+
+        [YamlMember(Alias = "stopParentProcessFirst")]
+        public bool StopParentProcessFirst { get; set; }
+        
+        [YamlMember(Alias = "resetFailureAfter")]
+        public TimeSpan ResetFailureAfter { get; set; }
+
+        [YamlMember(Alias = "stopTimeout")]
+        public TimeSpan StopTimeout { get; set; }
+        
+        [YamlMember(Alias = "startMode")]
+        public StartMode StartMode { get; set; }
+
+        [YamlMember(Alias = "serviceDependencies")]
+        public string[] ServiceDependencies { get; set; }
+
+        [YamlMember(Alias = "waitHint")]
+        public TimeSpan WaitHint { get; set; }
+
+        [YamlMember(Alias = "sleepTime")]
+        public TimeSpan SleepTime { get; set; }
+
+        [YamlMember(Alias = "interactive")]
+        public bool Interactive { get; set; }
+
+        [YamlMember(Alias = "priority")]
+        public ProcessPriorityClass Priority { get; set; }
+
+        [YamlMember(Alias = "beepOnShutdown")]
+        public bool BeepOnShutdown { get; set; }
+
+        [YamlMember(Alias = "environmentVariables")]
+        public Dictionary<string, string> EnvironmentVariables { get; set; }
+
+        public SC_ACTION[] FailureActions { get; set; }
+
+        [YamlMember(Alias = "failureActions")]
+        public List<YAML_SC_ACTION> YamlFailureActions { get; set; }
+
+        
+        public struct YAML_SC_ACTION
```

Comment:
I would recommend to use a class for that. Structs are easy, but classes might be helpful in the future to add methods and to refactor the codebase. Also, struct/class name should be changed IMHO